capsules/src/touch.rs

```diff
@@ -102,13 +102,26 @@ impl<'a> Touch<'a> {
                 break;
             }
         }
-        self.touch.map_or(Err(ErrorCode::NODEVICE), |touch| {
-            if enabled {
-                touch.enable()
-            } else {
-                touch.disable()
-            }
-        })
+        self.touch.map_or(
+            // if there is no single touch device
+            // try to use the multi touch device and
+            // report only a single touch
+            self.multi_touch
+                .map_or(Err(ErrorCode::NODEVICE), |multi_touch| {
+                    if enabled {
+                        multi_touch.enable()
+                    } else {
+                        multi_touch.disable()
+                    }
+                }),
```

Comment:
The first argument passed to `Option::map_or()` is eagerly evaluated (see https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or), so this code will run regardless of whether `touch.is_some()`. To get the behavior I think you want here, you should use `map_or_else()`.

Reply:
I think the documentation for `map_or` needs to be fixed. The closure is *not* eagerly evaluated (see [the implementation](https://doc.rust-lang.org/src/core/option.rs.html#514)), so this will work as intended. The "default" argument *is* eagerly evaluated in that it is computed before `map_or` is invoked.

Reply:
For the outer 'map_or()' call ('self.touch.map_or()'), the "default argument" in this code is an expression with side effects. So the default argument being evaluated before the call to map_or makes those side effects happen regardless. Unless I am misunderstanding?

Reply:
Oh, I didn't see the outer `map_or` call. Disregard my previous comment.

Reply:
I made a beginner's mistake 😄  I absolutely overlooked the fact that the function call is a parameter.